compiler/rustc_ty/src/ty.rs

```diff
@@ -260,6 +263,11 @@ fn param_env(tcx: TyCtxt<'_>, def_id: DefId) -> ty::ParamEnv<'_> {
     // are any errors at that point, so after type checking you can be
     // sure that this will succeed without errors anyway.
 
+    if tcx.sess.opts.debugging_opts.chalk {
```

Comment:
If new perf isn't good, I suspect this is the problem and we need to move this back to chalk_fulfill.